codec-dns/src/main/java/io/netty/handler/codec/dns/DatagramDnsQuery.java

```diff
@@ -0,0 +1,148 @@
+        this(sender, recipient, id, DnsOpCode.QUERY);
+    }
+
+    public DatagramDnsQuery(
+            InetSocketAddress sender, InetSocketAddress recipient, int id, DnsOpCode opCode) {
+        super(id, opCode);
+        this.sender = sender;
+        this.recipient = recipient;
+    }
+
+    @Override
+    public DatagramDnsQuery content() {
+        return this;
+    }
+
+    @Override
+    public InetSocketAddress sender() {
+        return sender;
+    }
+
+    @Override
+    public InetSocketAddress recipient() {
+        return recipient;
+    }
+
+    @Override
+    public DatagramDnsQuery setId(int id) {
+        return (DatagramDnsQuery) super.setId(id);
+    }
+
+    @Override
+    public DatagramDnsQuery setOpCode(DnsOpCode opCode) {
+        return (DatagramDnsQuery) super.setOpCode(opCode);
+    }
+
+    @Override
+    public DatagramDnsQuery setRecursionDesired(boolean recursionDesired) {
+        return (DatagramDnsQuery) super.setRecursionDesired(recursionDesired);
+    }
+
+    @Override
+    public DatagramDnsQuery setZ(int z) {
+        return (DatagramDnsQuery) super.setZ(z);
+    }
+
+    @Override
+    public DatagramDnsQuery setQuestion(DnsQuestion question) {
+        return (DatagramDnsQuery) super.setQuestion(question);
+    }
+
+    @Override
+    public DatagramDnsQuery setAdditionalRecord(DnsRecord record) {
+        return (DatagramDnsQuery) super.setAdditionalRecord(record);
+    }
+
+    @Override
+    public DatagramDnsQuery touch() {
+        return (DatagramDnsQuery) super.touch();
+    }
+
+    @Override
+    public DatagramDnsQuery touch(Object hint) {
+        return (DatagramDnsQuery) super.touch(hint);
+    }
+
+    @Override
+    public DatagramDnsQuery retain() {
+        return (DatagramDnsQuery) super.retain();
+    }
+
+    @Override
+    public DatagramDnsQuery retain(int increment) {
+        return (DatagramDnsQuery) super.retain(increment);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!super.equals(obj)) {
+            return false;
+        }
+
+        if (!(obj instanceof AddressedEnvelope)) {
+            return false;
+        }
+
+        @SuppressWarnings("unchecked")
+        final AddressedEnvelope<?, SocketAddress> that = (AddressedEnvelope<?, SocketAddress>) obj;
+        if (sender() == null) {
+            if (that.sender() != null) {
+                return false;
+            }
+        } else if (!sender().equals(that.sender())) {
+            return false;
+        }
+
+        if (recipient() == null) {
+            if (that.recipient() != null) {
+                return false;
+            }
+        } else if (!recipient().equals(that.recipient())) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int hashCode = super.hashCode();
+        if (sender() != null) {
+            hashCode = hashCode * 31 + sender().hashCode();
```

Comment:
We use `31` as the prime for hash codes in a bunch of places...in other classes I have made a private static define for this...I'm wondering should we extract this prime to somewhere more accessible?